src/uct/sm/mm/posix/mm_posix.c

```diff
@@ -122,15 +131,19 @@ uct_posix_md_query(uct_md_h tl_md, uct_md_attr_v2_t *md_attr)
     const uct_posix_md_config_t *posix_config =
                     ucs_derived_of(md->config, uct_posix_md_config_t);
     struct statvfs shm_statvfs;
+    size_t shm_size;
 
     if (statvfs(posix_config->dir, &shm_statvfs) < 0) {
         ucs_error("could not stat shared memory device %s (%m)",
                   UCT_POSIX_SHM_OPEN_DIR);
         return UCS_ERR_NO_DEVICE;
     }
 
-    uct_mm_md_query(&md->super, md_attr,
-                    shm_statvfs.f_bsize * shm_statvfs.f_bavail);
+    shm_size = shm_statvfs.f_bsize * shm_statvfs.f_bavail;
+    uct_mm_md_query(&md->super, md_attr, shm_size);
+    if (shm_size < posix_config->shm_min_size) {
+        md_attr->alloc_mem_types = 0;
+    }
```

Comment:
there is a better way to do it by passing appropriate parameter to uct_mm_md_query as "max_alloc"